src/context/EarnManagerProvider/providers/yearn/components/YearnManager/Withdraw/YearnWithdraw.tsx

```diff
@@ -0,0 +1,448 @@
+import { EarnParams, EarnQueryParams } from 'context/EarnManagerProvider/EarnManagerProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { useFlattenedBalances } from 'hooks/useBalances/useFlattenedBalances'
+import { useFetchAsset } from 'hooks/useFetchAsset/useFetchAsset'
+import { useMarketData } from 'hooks/useMarketData/useMarketData'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { poll } from 'lib/poll/poll'
+
+import { YearnVault, YearnVaultApi } from '../../../api/api'
+import { YearnRouteSteps } from '../../YearnRouteSteps'
+import { initialState, reducer, YearnWithdrawActionType } from './WithdrawReducer'
+
+enum WithdrawPath {
+  Withdraw = '/withdraw',
+  Confirm = '/withdraw/confirm',
+  ConfirmSettings = '/withdraw/confirm/settings',
+  Status = '/withdraw/status'
+}
+
+export const routes = [
+  { step: 0, path: WithdrawPath.Withdraw, label: 'Withdraw Amount' },
+  { step: 1, path: WithdrawPath.Confirm, label: 'Confirm Withdraw' },
+  { path: WithdrawPath.ConfirmSettings, label: 'Confirm Settings' },
+  { step: 2, path: WithdrawPath.Status, label: 'Status' }
+]
+
+type YearnWithdrawProps = {
+  api: YearnVaultApi
+}
+
+// TODO: Remove when vaults are added to asset service
+const makeVaultAsset = (vault: YearnVault): Asset => {
+  return {
+    chain: ChainTypes.Ethereum,
+    color: '#FFFFFF',
+    contractType: ContractTypes.ERC20,
+    explorer: 'https://etherscan.io',
+    explorerTxLink: 'https://etherscan.io/tx/',
+    icon: vault.icon,
+    name: vault.name,
+    network: NetworkTypes.MAINNET,
+    precision: vault.decimals,
+    receiveSupport: true,
+    secondaryColor: '#FFFFFF',
+    sendSupport: true,
+    slip44: 60,
+    symbol: vault.symbol,
+    tokenId: vault.address,
+    caip19: caip19.toCAIP19({
+      chain: ChainTypes.Ethereum,
+      network: NetworkTypes.MAINNET,
+      tokenId: vault.address,
+      contractType: ContractTypes.ERC20
+    })
+  }
+}
+
+export const YearnWithdraw = ({ api }: YearnWithdrawProps) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const { query, history: browserHistory } = useBrowserRouter<EarnQueryParams, EarnParams>()
+  const { chain, contractAddress: vaultAddress, tokenId } = query
+
+  // Asset info
+  const asset = useFetchAsset({ chain, tokenId })
+  const marketData = useMarketData({ chain, tokenId })
+  const feeAsset = useFetchAsset({ chain })
+  const feeMarketData = useMarketData({ chain })
+
+  // user info
+  const chainAdapterManager = useChainAdapters()
+  const chainAdapter = chainAdapterManager.byChain(ChainTypes.Ethereum)
+  const { state: walletState } = useWallet()
+  const { balances, loading } = useFlattenedBalances()
+
+  // navigation
+  const memoryHistory = useHistory()
+  const location = useLocation()
+  const WithdrawRoute = matchPath(location.pathname, { path: WithdrawPath.Withdraw, exact: true })
```

Comment:
nit pick can we lowercase this var name? withdrawRoute

Reply:
done.